src/Exadel.CrazyPrice/Exadel.CrazyPrice.Common/CrazyPriceHost.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Exadel.CrazyPrice.Common.Configurations;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.IO;
+
+namespace Exadel.CrazyPrice.Common
+{
+    public static class CrazyPriceHost
+    {
+        public static void CheckLoggerAndStartHost(IHostBuilder hostBuilder,
+            string initMessage, string errorInitMessage)
+        {
+            const string useLogging = "Logging:UseLogging";
+            const string loggerName = "Logging:LoggerName";
+            const string nLog = "nlog";
+            const string serilog = "serilog";
+            const string appsettings = "appsettings.json";
+
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile(appsettings)
+                .Build();
+
+            if (!bool.TryParse(configuration.GetSection(useLogging).Value, out var hasLogging))
+            {
+                throw new ArgumentException(useLogging);
+            }
+
+            if (!hasLogging)
+            {
+                hostBuilder.Build().Run();
+                return;
+            }
+
+            var usedLogger = configuration.GetSection(loggerName).Value.ToLowerInvariant();
+
+            ICrazyPriceLogger logger = usedLogger switch
```

Comment:
Rework to static instances - no need for interfaces here

Reply:
Removed.